code/modules/atmospherics/machinery/pipes/pipes.dm

```diff
@@ -1,15 +1,22 @@
 /obj/machinery/atmospherics/pipe
-	icon = 'icons/obj/pipes_n_cables/pipes_bitmask.dmi'
+	icon = 'icons/obj/pipes_n_cables/!pipes_bitmask.dmi'
 	damage_deflection = 12
-	var/datum/gas_mixture/air_temporary //used when reconstructing a pipeline that broke
+	/// Temporary holder for gases in the absence of a pipeline
+	var/datum/gas_mixture/air_temporary
+
+	/// The gas capacity this pipe contributes to a pipeline
 	var/volume = 0
 
 	use_power = NO_POWER_USE
 	can_unwrench = 1
+	/// The pipeline this pipe is a member of
 	var/datum/pipeline/parent = null
 
 	paintable = TRUE
 
+	/// Determines if this pipe will be given gas visuals
+	var/has_gas_visuals = TRUE
```

Comment:
this should be opt-in instead.

Reply:
Except that this isn't an abstract type and can be used by itself with gas visuals

Reply:
tfw your leaf is also a branch

Reply:
But you won't see this object in game, so it is an "abstract" type.. Kind of